### PR TITLE
[MM-53192] Patch Show Full Name issue in Insights team_members API

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This PR updates the logic to make sure to check for the "Show Full Name" option and if it is disabled, do not include the first and last name in the response.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53192

```release-note
Fix an issue where an Insights API would show data that a user may not have access to.
```

### CODE REVIEW

File: server/channels/app/team.go [hunk 1]
```diff
@@ -2149,7 +2149,7 @@ func (a *App) GetNewTeamMembersSince(c request.CTX, teamID string, opts *model.I
 		return nil, 0, model.NewAppError("GetNewTeamMembersSince", "app.insights.feature_disabled", nil, "", http.StatusNotImplemented)
 	}
 
-	ntms, count, err := a.Srv().Store().Team().GetNewTeamMembersSince(teamID, opts.StartUnixMilli, opts.Page*opts.PerPage, opts.PerPage)
+	ntms, count, err := a.Srv().Store().Team().GetNewTeamMembersSince(teamID, opts.StartUnixMilli, opts.Page*opts.PerPage, opts.PerPage, *a.Config().PrivacySettings.ShowFullName || a.SessionHasPermissionTo(*c.Session(), model.PermissionManageSystem))
```
Comment (isacikgoz) [inline server/channels/app/team.go:2152]: nit: Just to improve readability maybe we can use an allocation here:

```suggestion
	showFullName := *a.Config().PrivacySettings.ShowFullName || a.SessionHasPermissionTo(*c.Session(), model.PermissionManageSystem)
	ntms, count, err := a.Srv().Store().Team().GetNewTeamMembersSince(teamID, opts.StartUnixMilli, opts.Page*opts.PerPage, opts.PerPage, showFullName)
```

**Review by isacikgoz — APPROVED** [review]
LGTM, just a non-bloking thing.

File: server/channels/app/team.go (hunk 1 above)
Comment (harshilsharma63) [reply server/channels/app/team.go:2152]: +1 for @isacikgoz 's suggestion

**Review by harshilsharma63 — APPROVED** [review]
LGTM 👍

**Review by esarafianou — APPROVED** [review]
LGTM!

**Pull request merged**